irohad/ametsuchi/impl/burrow_storage.cpp

```diff
@@ -0,0 +1,140 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "ametsuchi/impl/burrow_storage.h"
+
+#include <cstddef>
+#include <cstring>
+#include <optional>
+#include <string_view>
+#include <type_traits>
+#include <vector>
+
+#include "ametsuchi/burrow_storage.hpp"
+#include "common/result.hpp"
+
+namespace {
+  inline void clearCharBuffer(Iroha_CharBuffer &buf) {
+    buf.data = nullptr;
```

Comment:
Where does the memory allocated for `buf.data` get released if not here?

Reply:
this method is used only for structure initialization and has nothing to do with previous contents. the memory originating in Go code should be managed there in my understanding. (imagine static const error strings vs dynamic ones).